arraycontext/context.py

```diff
@@ -304,25 +304,26 @@ def zeros_like(self, ary: Array) -> Array:
 
     @abstractmethod
     def from_numpy(self,
-                   array: Union["np.ndarray[Any, Any]", ScalarLike]
-                   ) -> Union[Array, ScalarLike]:
+                   array: ArrayOrContainerOrScalarT
+                   ) -> ArrayOrContainerOrScalarT:
```

Comment:
That type annotation is not correct, because it promises that the same types come back as go in, which is precisely the opposite of what `from_numpy` does. Also, I kind of like that `ndarray` was part of the type signature.

Reply:
Updated the annotations in https://github.com/inducer/arraycontext/compare/bb4d8fce5eec09fc59b5e125573f4ce402523ec2..8c9952aa46a6749a54330e6984d83f76c2249d7a

What do you think?